modules/atomic-widgets/elements/atomic-svg/atomic-svg.php

```diff
@@ -109,11 +113,52 @@ protected function render() {
 		echo $svg_html;
 	}
 
-	private function set_svg_attributes( \WP_HTML_Tag_Processor $svg, $settings ) {
-		$svg->set_attribute( 'fill', 'currentColor' );
-		$string_classes = implode( ' ', $settings['classes'] );
-		$svg->add_class( $string_classes );
-		$base_styles = $this->get_base_styles_dictionary()[ self::BASE_STYLE_KEY ];
-		$svg->add_class( $base_styles );
+	private function get_svg_content( $settings ) {
+
+		if ( isset( $settings['svg']['id'] ) ) {
+			$content = Utils::file_get_contents(
+				get_attached_file( $settings['svg']['id'] )
+			);
+
+			if ( $content ) {
+				return $content;
+			}
+		}
+
+		if (
+			isset( $settings['svg']['url'] ) &&
+			static::get_default_svg( 'url' ) !== $settings['svg']['url']
+		) {
+			$content = wp_safe_remote_get(
+				$settings['svg']['url']
+			);
+
+			if ( ! is_wp_error( $content ) ) {
+				return $content['body'];
+			}
+		}
+
+		$content = Utils::file_get_contents(
+			static::get_default_svg( 'path' )
+		);
+
+		return $content ? $content : null;
+	}
+
+	/**
+	 * @param string $context 'url' | 'path'
+	 *
+	 * @return string
+	 */
+	private static function get_default_svg( string $context ) {
```

Comment:
Well, technically:

```suggestion
	 * @return string | null
	 */
	private static function get_default_svg( string $context ) {
```

and you can also:

```suggestion
	private static function get_default_svg( string $context ): ?string {
```

But I this whole function might be deleted based on one of my previous comments, so... 🤷

Reply:
Deleted